assets/js/modules/ads/components/common/WooCommerceRedirectModal.test.js

```diff
@@ -167,6 +226,86 @@ describe( 'WooCommerceRedirectModal', () => {
 		expect( onBeforeSetupCallback ).toHaveBeenCalled();
 	} );
 
+	it( 'clicking "Continue with Site Kit" should trigger the correct internal tracking event when only WooCommerce is active', async () => {
+		registry.dispatch( MODULES_ADS ).receiveModuleData( {
+			plugins: {
+				[ PLUGINS.WOOCOMMERCE ]: {
+					active: true,
+				},
+				[ PLUGINS.GOOGLE_FOR_WOOCOMMERCE ]: {
+					active: false,
+					adsConnected: false,
+				},
+			},
+		} );
+
+		fetchMock.postOnce( moduleActivationEndpoint, {
+			body: { success: true },
+		} );
+		fetchMock.getOnce( userAuthenticationEndpoint, {
+			body: { needsReauthentication: false },
+		} );
+
+		const { getByText, waitForRegistry } = render( <ModalComponent />, {
+			registry,
+			viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
+		} );
+		await waitForRegistry();
+
+		const continueWithSiteKitButton = getByText(
+			/continue with site kit/i
+		);
+		fireEvent.click( continueWithSiteKitButton );
+
+		await waitForRegistry();
```

Comment:
We can't reuse `waitForRegistry` like this – or at least it won't work as expected. Have a look in `createWaitForRegistry`, but it's essentially single-use. Once the function is called, it awaits until state updates stop happening for some time and then unsubscribes. Once that happens, subsequent calls will essentially await a fulfilled promise every time. For `fireEvent` it should probably be wrapped in `act`. If we really needed to, we could create a new `waitForRegistry` to await after the click (within `act`) but that probably isn't necessary?

Reply:
This too was borrowed from existing implementations in this file, I will have to explore alternatives if this is not acceptable.